engine/gamesys/src/gamesys/gamesys_private.cpp

```diff
@@ -91,39 +98,57 @@ namespace dmGameSystem
                     out_desc.m_ElementIds[3] = element_ids[3];
                 }
 
-                if (value != 0x0 && use_value_ptr)
+                if (value != 0x0)
                 {
-                    out_desc.m_ValuePtr = (float*)value;
                     out_desc.m_Variant = dmGameObject::PropertyVar(*value);
+                    if (use_value_ptr)
+                    {
+                        // TODO: Make this more robust. If the constant is e.g. animated (which might get the pointer)
+                        // and then the memory is reallocated (E.g. constant value array grew due to newly set values),
+                        // then crashes could occur. Or what if it's animated, and then we reset the constant?
+                        out_desc.m_ValuePtr = (float*)value;
```

Comment:
This is a real concern.
We've had this for a while now, since we (I) introduced the render constants struct/paradigm in order to reduce memory usage.

I'm not entirely sure how to fix it yet.
Perhaps by cancelling an animation (somehow) if we set a new value to the constant.

In any case, I think we have some time to think about it as noone has reported the issue (yet).